### PR TITLE
Dragonflye: version update and  polish the draft genome with short-reads.

### PR DESCRIPTION
**Pull request by Daniel-VM**
## PR description
[Dragonflye](https://github.com/rpetit3/dragonflye) allows to polish the assembled genome with short reads, when the these are provided. However the current version of  `nf-core/dragonflye `doesn't allow that. 

I add in this PR: 
- An update of the dragonflye module version.
- A refactoring of the code to allow the dragonflye module to polish the draft genome with short reads
- A new nf-test for dragonflye hybrid-mode.

## PR checklist

Closes #XXX 

- [X] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/dragonflye/meta.yml [hunk 1]
```diff
@@ -16,6 +16,11 @@ input:
       description: |
         Groovy Map containing sample information
         e.g. [ id:'test', single_end:false ]
+  - reads:
```
Comment (rpetit3) [inline modules/nf-core/dragonflye/meta.yml:19]:
```suggestion
  - shortreads:
```

File: modules/nf-core/dragonflye/meta.yml [hunk 2]
```diff
@@ -16,6 +16,11 @@ input:
...
+      type: file
+      description: |
+        Optional. List of FastQ files of short reads (paired-end data) that will be used to polish the draft genome.
+      pattern: "*.fastq.gz"
   - reads:
```
Comment (rpetit3) [inline modules/nf-core/dragonflye/meta.yml:24]:
```suggestion
  - longreads:
```

**Comment by Daniel-VM** [comment]
Thanks @rpetit3 for your suggestions, I have incorporate them into [09212f1](https://github.com/nf-core/modules/pull/4316/commits/09212f1db1abf3f5cf92640d9edb26ed84ed01a6).  

Lets see if the `nf-test`  fix works 😅

**Comment by Daniel-VM** [comment]
Tests passed, but the review still pending approval @rpetit3 .

**Review by rpetit3 — APPROVED** [review]

**Pull request merged**